src/odemis/gui/cont/microscope.py

```diff
@@ -1441,3 +1441,84 @@ def _update_load_time(self, future, start, end):
+        self._tab_data = tab_data
+        self._tab_panel = tab_panel
+
+        # E-beam management
+        tooltips = {0: "Truen E-beam on", 1: "Turn E-beam off"}
+        self._ebeam_btn_ctrl = HardwareButtonController(tab_panel.btn_ebeam,
+                                                        self._main_data.ebeam_state,
+                                                        tooltips)
+        self._main_data.ebeam_state.subscribe(self._on_ebeam_state)
+        self._main_data.ebeam.power.subscribe(self._on_ebeam_power)
+
+        # Chamber management
+        self._press_btn_ctrl = ChamberButtonController(tab_panel.btn_pressure,
+                                                       self._main_data.chamberState,
+                                                       self._main_data)
+        self._main_data.chamberState.subscribe(self.on_chamber_state)
+        ch_pos = self._main_data.chamber.position
+        ch_pos.subscribe(self.on_chamber_pos, init=True)
+
+    def _on_ebeam_state(self, _):
+        # power is BooleanVA, ebeam_state IntVA (0 or 1)
+        self._main_data.ebeam.power.value = True if self._main_data.ebeam_state.value else False
+
+    def _on_ebeam_power(self, power):
+        # power is BooleanVA, ebeam_state IntVA (0 or 1)
+        self._main_data.ebeam_state.value = 1 if power else 0
+
+    def on_chamber_state(self, state):
+        """
+        Set the desired pressure on the chamber when the chamber's state changes
+        Only 'active' states (i.e. either CHAMBER_PUMPING or CHAMBER_VENTING)
+        will start a change in pressure.
+        """
+        logging.debug("Chamber state changed to %d", state)
+        chamber = self._main_data.chamber
+        if state == CHAMBER_PUMPING:
+            wx.CallAfter(self._press_btn_ctrl.Enable, False)
+            f = chamber.moveAbs({"vacuum": 1})
+            f.add_done_callback(self._on_vacuum)
+        elif state == CHAMBER_VENTING:
+            wx.CallAfter(self._press_btn_ctrl.Enable, False)
+            f = chamber.moveAbs({"vacuum": 0})
+            f.add_done_callback(self._on_vented)
+
+    def _on_vacuum(self, f):
+        self._main_data.chamberState.value = CHAMBER_VACUUM
+        wx.CallAfter(self._press_btn_ctrl.Enable, True)
+
+    def _on_vented(self, f):
+        self._main_data.chamberState.value = CHAMBER_VENTED
+        wx.CallAfter(self._press_btn_ctrl.Enable, True)
+
+    def on_chamber_pos(self, position):
+        """
+        Determine the state of the chamber when the pressure changes on the hardware.
+        """
+        if self._main_data.chamberState.value in (CHAMBER_PUMPING, CHAMBER_VENTING):
+            # .position VA only knows VACUUM/VENTED, don't update when it's PUMPING/VENTING, we keep
+            # track of this here and in the ChamberButtonController
+            return
+
+        if position["vacuum"] == 0:
+            self._main_data.chamberState.value = CHAMBER_VENTED
+        elif position["vacuum"] == 1:
```

Comment:
Please, don't hard-code to 0 and 1. Instead, look at the axis choices, and map the value linked to "vented" and "vacuum" to the values you're looking for.